cub/cub/device/device_reduce.cuh

```diff
@@ -1819,6 +2675,203 @@ public:
                          static_cast<OffsetT>(num_items),
                          stream);
   }
+
+  //! @rst
+  //! Reduces segments of values, where segments are demarcated by corresponding runs of identical keys.
+  //!
+  //! This operation computes segmented reductions within ``d_values_in`` using the specified binary ``reduction_op``
+  //! functor. The segments are identified by "runs" of corresponding keys in `d_keys_in`, where runs are maximal
+  //! ranges of consecutive, identical keys. For the *i*\ :sup:`th` run encountered, the first key of the run and
+  //! the corresponding value aggregate of that run are written to ``d_unique_out[i]`` and ``d_aggregates_out[i]``,
+  //! respectively. The total number of runs encountered is written to ``d_num_runs_out``.
+  //!
+  //! - The ``==`` equality operator is used to determine whether keys are equivalent
+  //! - Provides determinism based on the environment's determinism requirements.
+  //!   To request "run-to-run" determinism, pass `cuda::execution::require(cuda::execution::determinism::run_to_run)`
+  //!   as the `env` parameter.
+  //!   "gpu-to-gpu" determinism is not supported for `ReduceByKey` at the moment.
```

Comment:
critical: if this is not enforced, users will pass gpu-to-gpu determinism and expect it to work. Please, add static assert.